### PR TITLE
[IndexTable] Fix header cutting off loading banner due to z-index

### PR DESCRIPTION
**Pull request by stefanlegg**
### WHY are these changes introduced?

Related to: https://github.com/Shopify/polaris/pull/8277
In order to support some tooltip behavior the z-index changed for table headers. Loading banners need to have their z-index moved up to not be cut off by the header.

Example issue:
Screenshot 2023-02-15 at 9 42 03 AM

### WHAT is this pull request doing?

This PR removes `--pc-index-table-loading-panel` and the loading banner will now use `--p-z-2` for z-index instead.

Example with fix:
Screenshot 2023-02-15 at 9 42 21 AM

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by stefanlegg** [comment]
[note] Checked this z-index against sticky headers to make sure those dont overlap either
Screenshot 2023-02-15 at 10 28 49 AM

File: polaris-react/src/components/IndexTable/IndexTable.scss [hunk 1]
```diff
@@ -495,7 +494,7 @@ $loading-panel-height: 53px;
   right: 0;
   visibility: hidden;
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-  z-index: var(--pc-index-table-loading-panel);
+  z-index: var(--p-z-2);
```
Comment (LauraAubin) [inline polaris-react/src/components/IndexTable/IndexTable.scss]: Why is this change to `StickyTable` needed?

Comment (stefanlegg) [reply polaris-react/src/components/IndexTable/IndexTable.scss]: I thought this had the same sort of problem, but after double checking I guess not. I'm reverting this change and just renaming `--pc-index-table-loading-panel` instead.

**Review by LauraAubin — APPROVED** [review]

**Review by ChrisMLee — APPROVED** [review]

**Pull request merged**